src/handlers.ts

```diff
@@ -129,15 +129,16 @@ function onEnter(editor: Editor, ev: KeyboardEvent): void {
         return ev.preventDefault()
     }
 
-    const [caption, captionPath] = Editor.above(editor, { match: Caption.isCaption }) ?? []
+    const [caption] = Editor.above(editor, { match: Caption.isCaption }) ?? []
     if (caption != null) {
         Editor.withoutNormalizing(editor, () => {
             if (!Range.isCollapsed(selection)) {
                 Editor.deleteFragment(editor)
             }
 
-            Transforms.splitNodes(editor)
-            Transforms.liftNodes(editor, { at: Path.next(captionPath!) })
+            if (!ev.shiftKey) {
+                Editor.insertText(editor, '\n')
```

Comment:
A new line (`'\n'`) in this context will be normalized to a space, so this change effectively makes enter and space equivalent.

What the original code intended to do, when an enter is pressed inside a caption, is split the caption in two and move the part after the cursor out of the figure. The important part is moving out of figure, not the visual appearance of a break.